apps/opik-backend/src/main/resources/liquibase/db-app-state/migrations/000004_add_prompt_library_tables.sql

```diff
@@ -0,0 +1,33 @@
+--liquibase formatted sql
+--changeset thiagohora:add_prompt_library_tables
+
+CREATE TABLE prompts (
+    id CHAR(36) NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    description VARCHAR(255),
+    created_at TIMESTAMP(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
+    created_by VARCHAR(100) NOT NULL DEFAULT 'admin',
+    last_updated_at TIMESTAMP(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),
+    last_updated_by VARCHAR(100) NOT NULL DEFAULT 'admin',
+    workspace_id VARCHAR(150) NOT NULL,
+    CONSTRAINT `prompts_pk` PRIMARY KEY (id),
+    CONSTRAINT `prompts_workspace_id_name_uk` UNIQUE (workspace_id, name)
+);
+
+CREATE TABLE prompt_versions (
+    id CHAR(36) NOT NULL,
+    prompt_id CHAR(36) NOT NULL,
+    commit VARCHAR(7) NOT NULL,
+    template MEDIUMTEXT NOT NULL,
+    created_at TIMESTAMP(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
+    created_by VARCHAR(100) NOT NULL DEFAULT 'admin',
+    last_updated_at TIMESTAMP(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),
+    last_updated_by VARCHAR(100) NOT NULL DEFAULT 'admin',
+    workspace_id VARCHAR(150) NOT NULL,
+    CONSTRAINT `prompt_versions_pk` PRIMARY KEY (id),
+    CONSTRAINT `prompt_versions_prompt_id_version_uk` UNIQUE (prompt_id, commit),
+    CONSTRAINT `prompt_versions_prompt_id_fk` FOREIGN KEY (prompt_id) REFERENCES prompts(id)
```

Comment:
Minor: not a big fan of foreign keys, as in the long run and for bigger applications it's better to handle that logic on the application side. You could do actions on update or delete, but better not for the same reason. With all this in mind, probably better to remove the foreign key. But this is ok to stay as well.

Reply:
I was wondering the same. I will remove it since we already have an index starting with it

Reply:
Changed